contrib/log/slog/slog.go

```diff
@@ -31,21 +39,71 @@ func NewJSONHandler(w io.Writer, opts *slog.HandlerOptions) slog.Handler {
 
 // WrapHandler enhances the given logger handler attaching tracing information to logs.
 func WrapHandler(h slog.Handler) slog.Handler {
-	return &handler{h}
+	return &handler{wrapped: h}
 }
 
 type handler struct {
-	slog.Handler
+	wrapped slog.Handler
+	groups  []group
+}
+
+// Enabled calls the wrapped handler Enabled method.
+func (h *handler) Enabled(ctx context.Context, level slog.Level) bool {
+	return h.wrapped.Enabled(ctx, level)
 }
 
 // Handle handles the given Record, attaching tracing information if found.
 func (h *handler) Handle(ctx context.Context, rec slog.Record) error {
+	reqHandler := h.wrapped
+
+	// We need to ensure the trace id and span id keys are set at the root level:
+	// https://docs.datadoghq.com/tracing/other_telemetry/connect_logs_and_traces/
+	// In case the user has created group loggers, we ignore those and
+	// set them at the root level.
 	span, ok := tracer.SpanFromContext(ctx)
 	if ok {
-		rec.Add(
-			slog.Uint64(ext.LogKeyTraceID, span.Context().TraceID()),
-			slog.Uint64(ext.LogKeySpanID, span.Context().SpanID()),
-		)
+		traceID := strconv.FormatUint(span.Context().TraceID(), 10)
+		spanID := strconv.FormatUint(span.Context().SpanID(), 10)
+
+		attrs := []slog.Attr{
+			slog.String(ext.LogKeyTraceID, traceID),
+			slog.String(ext.LogKeySpanID, spanID),
+		}
+		reqHandler = reqHandler.WithAttrs(attrs)
+	}
+	for _, g := range h.groups {
+		reqHandler = reqHandler.WithGroup(g.name)
+		if len(g.attrs) > 0 {
+			reqHandler = reqHandler.WithAttrs(g.attrs)
+		}
+	}
```

Comment:
So everytime `Handle` is invoked, we have to copy over the existing attributes + the new trace attributes, as well as the existing groups?
Is this a performance concern? Is `Handle` called often?